--- a/threads.c	2021-06-18 02:10:43.270585932 -0700
+++ b/threads.c	2021-06-18 02:10:34.059601330 -0700
@@ -49,7 +49,7 @@
 static int libxml_is_threaded = -1;
 #ifdef __GNUC__
 #ifdef linux
-#if (__GNUC__ == 3 && __GNUC_MINOR__ >= 3) || (__GNUC__ > 3)
+#if (__GNUC__ == 3 && __GNUC_MINOR__ >= 3) || (__GNUC__ > 3 && __GNUC__ < 8)
 extern int pthread_once (pthread_once_t *__once_control,
                          void (*__init_routine) (void))
 	   __attribute((weak));
